Jenkinsfile

```
@@ -12,18 +14,20 @@ def runTests(name, bedrockCommand, runAll) {
   }
 
   def command = runAll ? bedrockCommand + ' --ignore-lerna-changed=true' : bedrockCommand
-  def successfulTests = execHandle(command)
+  def testStatus = exec(script: command, returnStatus: true)
 
-  echo "Writing JUnit results for " + name + " on node: $NODE_NAME"
+  echo "Writing JUnit results for ${name} on node: $NODE_NAME"
   junit allowEmptyResults: true, testResults: 'scratch/TEST-*.xml'
 
-  if (!successfulTests) {
-    echo "Tests failed for " + name + " so passing failure as exit code for node: $NODE_NAME"
-    exec("exit 1")
+  // If the tests failed (exit code 4) then just mark it as unstable
+  if (testStatus == 4) {
+    unstable("Tests failed for ${name}")
+  } else if (testStatus != 0) {
+    error("Unexpected error running tests for ${name} so passing failure as exit code")
```

Maybe change it to this since the old code had a exit 1 so then it kind of described what the code below it did. Now it's just an error described by the function call it self.

```suggestion
    error("Unexpected error running tests for ${name}")
```

---

It's just a 1:1 copy of waluigi atm. I wanted to keep it the same, but I don't have a strong attachment to that so if you still want to I'll change it here.

---

Yes but the message is like throw new Error('Error: something') kind of the same information twice an error call is kind of implied that it will produce an error code. But I have no strong opinion here you could just leave it as is just looked a bit funny.

---

I'm just going to leave it then so it's clear it's a copy. There's TINY-9313 for you all to remove the duplication anyways.